### PR TITLE
Fix issue with concurrent reads

### PR DESCRIPTION
**Pull request by ludeeus**
## Type of change

- [ ] Dependency upgrade
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (which adds functionality to the supervisor)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue:
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [x] The code has been formatted using Black (`black --fast supervisor tests`)
- [ ] Tests have been added to verify that the new code works.

If API endpoints of add-on configuration are added/changed:

- [ ] Documentation added/updated for [developers.home-assistant.io][docs-repository]

[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[docs-repository]: https://github.com/home-assistant/developers.home-assistant

### CODE REVIEW

File: supervisor/homeassistant/websocket.py [hunk 1]
```diff
@@ -30,6 +31,9 @@ def __init__(
 
     async def async_send_command(self, message: Dict[str, Any]):
         """Send a websocket command."""
+        while self.client._waiting is not None:  # pylint: disable=protected-access
+            await asyncio.sleep(0.001)
```
Comment (pvizeli) [inline supervisor/homeassistant/websocket.py:35]: Is a bit hacky, let's use https://docs.python.org/3/library/asyncio-sync.html#lock

File: supervisor/homeassistant/websocket.py [hunk 2]
```diff
@@ -24,29 +25,34 @@ def __init__(
...
+        self.client: aiohttp.ClientWebSocketResponse = client
+        self.message_id: int = 0
+        self._lock: asyncio.Lock = asyncio.Lock()
 
     async def async_send_command(self, message: Dict[str, Any]):
         """Send a websocket command."""
-        self.message_id += 1
-        message["id"] = self.message_id
+        while self._lock.locked():
+            await asyncio.sleep(0)
```
Comment (pvizeli) [inline supervisor/homeassistant/websocket.py:36]: Why not wait on the lock itself? we can also create it on the job, `@Job(limit=ExecutionLimit.SingleWait)` would be pretty easy to add into Job decorator and we can reuse it

Comment (pvizeli) [reply supervisor/homeassistant/websocket.py:36]: Before you close the PR: https://github.com/home-assistant/supervisor/pull/2612 - You can use that

Comment (ludeeus) [reply supervisor/homeassistant/websocket.py:36]: We can not and should not use Job here, it's a helper.
In the future all jobs will also use this helper to publish their updates, at that point we will end in a loop

**Review by pvizeli — APPROVED** [review]

**Pull request merged**